src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs

```diff
@@ -208,7 +209,14 @@ internal void BuildAndLogConsoleLog(string msg, LogLevel level)
 
         public void WaitForProcessExitInMilliSeconds(int waitTime)
         {
-            Process.WaitForExit(waitTime);
+            try
+            {
+                Process.WaitForExit(waitTime);
```

Comment:
Can we also check if the process has exited here first (in addition to the exception handling) in an attempt to avoid this condition?

Reply:
yes, that's easy. coming soon...